### PR TITLE
Fix gzip failure in current implementation of ExtendedJsonRpcApi

### PR DESCRIPTION
**Pull request by jseagrave21**
**What current issue(s) does this address, or what feature is it adding?**
Currently when a request is made with a header including support for gzip such as `Accept-Encoding: deflate, gzip;q=1.0, *;q=0.5`, this error occurs

```
======================================================================
ERROR: test_gzip_compression (neo.api.JSONRPC.test_extended_json_rpc_api.ExtendedJsonRpcApiTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/c/users/jseag/neo/neo-python-dev/neo/api/JSONRPC/test_extended_json_rpc_api.py", line 163, in test_gzip_compression
    res = self.app.home(mock_req)
  File "/mnt/c/users/jseag/neo/neo-python-dev/neo/api/utils.py", line 27, in wrapper
    response_data = gzip.compress(bytes(response_data, 'utf-8'), compresslevel=COMPRESS_FASTEST)
TypeError: encoding without a string argument

----------------------------------------------------------------------
```

This PR fixes this problem and maintains all other functionality.

**How did you solve this problem?**
Trial and Error

**How did you make sure your solution works?**
Unittest and personal testing 

**Are there any special changes in the code that we should be aware of?**
No

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [x] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

**Comment by jseagrave21** [comment]
From my research, it seems as though functions with decorators are not meant to be modified or extended. To this end, I am proposing a new idea: adding an identifier input `server_type` which can be altered as desired. My intention is to be able to identify RPC server types with an `OPTIONS` request where it would otherwise need to be accomplish through trial and error.

**Comment by jseagrave21** [comment]
This method allows for identification of server_type while maintaining all other functionality

File: neo/api/JSONRPC/JsonRpcApi.py [hunk 1]
```diff
@@ -172,12 +172,17 @@ def home(self, request):
...
         elif "OPTIONS" == request.method.decode("utf-8"):
-            return {'supported HTTP methods': ("GET", "POST"),
-                    'JSON-RPC server type': "default"}
+            return self.options_response()
 
         error = JsonRpcError.invalidRequest("%s is not a supported HTTP method" % request.method.decode("utf-8"))
         return self.get_custom_error_payload(request_id, error.code, error.message)
 
+    @classmethod
+    def options_response(self):
```
Comment (LysanderGG) [inline neo/api/JSONRPC/JsonRpcApi.py]: The argument should be `cls` rather than `self`

Comment (jseagrave21) [reply neo/api/JSONRPC/JsonRpcApi.py]: Nice! Thanks

Comment (LysanderGG) [reply neo/api/JSONRPC/JsonRpcApi.py]: Same in `neo/api/JSONRPC/ExtendedJsonRpcApi.py` 😅

Comment (jseagrave21) [reply neo/api/JSONRPC/JsonRpcApi.py]: I wondered about that but does that mean test_json_rpc_api.py is wrong here https://github.com/CityOfZion/neo-python/blob/e3780126f9cf8175f9f3c7ad27f36f2baae20508/neo/api/JSONRPC/test_json_rpc_api.py#L37?

Comment (LysanderGG) [reply neo/api/JSONRPC/JsonRpcApi.py]: Technically the parameter can be named whatever you want, but it's a convention to name it `cls` for class methods as it is a convention to name the first parameter `self` for instance methods.
https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments

**Comment by jseagrave21** [comment]
@LysanderGG Okay, I think I got them all. Thank you for the review!

**Pull request merged**